chips/earlgrey/src/chip.rs

```diff
@@ -16,7 +16,7 @@ use crate::plic::PLIC;
 
 pub struct EarlGrey<'a, A: 'static + Alarm<'static>, I: InterruptService<()> + 'a> {
     userspace_kernel_boundary: SysCall,
-    pmp: PMP<4, 2>,
+    pmp: PMP<8>,
```

Comment:
did we just have the wrong value here before?

Reply:
No, 4 is the number of regions that OT currently has.

The idea of bumping this up to 16 is that we can auto-detect the number between 0 and 16.

Reply:
Oh, I see. I could see this being misleading to people who might look at this file and assume OT has 16 regions. Perhaps we should rename that const generic variable?

Reply:
The comment in PMP should describe this

Any ideas on a better name?

Reply:
perhaps `MAX_AVAILABLE_REGIONS_OVER_TWO`? I realize that is a bit long.

Reply:
Done!

Reply:
The hardware should be fixed, and I think adding uncertainty in the name makes it confusing. OT is a unique case for now, but eventually the hardware will be fixed.

Reply:
I don't think it should be though.

The RISC-V spec specifically allows any number of PMP regions. It's entirely possible there could be different chips in the same family with different number of regions. Tock can determine the number available at run time.

There are also a few FPGA implementations supported in Tock which can be adjusted and also detected at run time.